### PR TITLE
NIFI-13722: Kerberos ticket renewal issue due static thread pool in Iceberg library (nifi-1.x)

### PR DESCRIPTION
**Pull request by mark-bathori**
# Summary

[NIFI-13722](https://issues.apache.org/jira/browse/NIFI-13722)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-common/src/main/java/org/apache/nifi/processors/iceberg/AbstractIcebergProcessor.java [hunk 1]
```diff
@@ -68,35 +70,34 @@ public abstract class AbstractIcebergProcessor extends AbstractProcessor impleme
...
 
         if (kerberosUserService != null) {
-            final KerberosUser kerberosUser = kerberosUserService.createKerberosUser();
-            kerberosUserReference.set(kerberosUser);
+            KerberosUser kerberosUser;
+            if (kerberosUserReference.get() == null) {
+                kerberosUser = kerberosUserService.createKerberosUser();
+            } else {
+                kerberosUser = kerberosUserReference.get();
+            }
```
Comment (dan-s1) [inline nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-common/src/main/java/org/apache/nifi/processors/iceberg/AbstractIcebergProcessor.java:91]: Just call `get()` once
```suggestion
            KerberosUser kerberosUser = kerberosUserReference.get() ;
            if (kerberosUser == null) {
                kerberosUser = kerberosUserService.createKerberosUser();
            } 
```

File: nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-common/src/main/java/org/apache/nifi/processors/iceberg/AbstractIcebergProcessor.java [hunk 2]
```diff
@@ -115,18 +116,26 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                 });
 
             } catch (Exception e) {
-                getLogger().error("Privileged action failed with kerberos user " + kerberosUser, e);
-                session.transfer(session.penalize(flowFile), REL_FAILURE);
+                if (!handleAuthErrors(e, session, context)) {
+                    getLogger().error("Privileged action failed with kerberos user " + kerberosUser, e);
```
Comment (dan-s1) [inline nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-common/src/main/java/org/apache/nifi/processors/iceberg/AbstractIcebergProcessor.java:120]: Use interpolation and not concatenation.
```suggestion
                    getLogger().error("Privileged action failed with kerberos user {}" , kerberosUser, e);
```

**Comment by mark-bathori** [comment]
Thanks @dan-s1 for the comments. This is a backport PR so these changes are already merged into the main branch. To keep the consistency between the two branch and PRs I would keep the changes as it is, but your requested changes could be done in a separate PR if necessary.

**Review by turcsanyip — APPROVED** [review]
As per @mark-bathori comment, it is a backport PR and I would also keep it as it is.

Thanks @dan-s1 for your review comments, I overlooked them in the original review on the main branch. We can address them in a follow-up PR or next time when we touch the Iceberg processors.

+1 Merging this for now.

**Comment by turcsanyip** [comment]
Merged.

**Pull request closed without merging**